joshua/joshua_agent.py

```diff
@@ -571,23 +606,38 @@ def _run_ensemble(
                 )
 
         # Write the output to the tmp directory so that it is picked up when we save (if we do so).
-        if should_save(retcode, save_on):
-            try:
-                to_write = os.path.join(where, "tmp", "console.log")
-
-                i = 0
-                while os.path.exists(to_write):
-                    to_write = os.path.join(where, "tmp", "console-{0}.log".format(i))
-                    i += 1
-
-                with open(to_write, "wb") as fout:
-                    fout.write(output)
+        del_output_file = not should_save(retcode, save_on)
+        to_write = os.path.join(where, "tmp", "console.log")
+        try:
```

Comment:
What's the motivation for unconditionally entering the block of code that was guarded by `should_save` previously?

Reply:
We now need the file -- so `should_save` will determine whether the file will get deleted or not. But `joshua_done` will expect the file to be present while it is executed